### PR TITLE
zsh-powerlevel10k: 1.0 → 1.2.0

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

New upstream release. Also updates gitstatusd and its internally used libgit2 fork.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

@mmlb @flokli @andir @maralorn

### CODE REVIEW

File: pkgs/shells/zsh/zsh-powerlevel10k/default.nix [hunk 1]
```diff
@@ -5,12 +5,12 @@
...
 stdenv.mkDerivation {
   pname = "powerlevel10k";
-  version = "1.1";
+  version = "1.2.0";
   src = fetchFromGitHub {
     owner = "romkatv";
     repo = "powerlevel10k";
-    rev = "9d9c50611da19044370ee759e593ccadbad32a6a";
-    sha256 = "0jl4jamh7i8w7lp3qbbakh1wsyrrmslsrjwh9jgyvw50a83hp6fj";
+    rev = "cb3f58d21f643d26d71f520d61a592816cca3edf";
```
Comment (cole-h) [inline pkgs/shells/zsh/zsh-powerlevel10k/default.nix]: Would it make sense to just substitute in the `version`? They both point to the same commit, but this way the `rev` shouldn't need to be updated every time.

```suggestion
    rev = "v${version}";
```

Comment (flokli) [reply pkgs/shells/zsh/zsh-powerlevel10k/default.nix]: I assume this was mostly a leftover from the time when powerlevel10k didn't do releases - we should indeed do as suggested here.

Comment (mweinelt) [reply pkgs/shells/zsh/zsh-powerlevel10k/default.nix]: Indeed, i pushed an update.

**Comment by maralorn** [comment]
I have tested this on my system. Seems to be working just fine. Thanks!

**Pull request merged**